overlord/snapstate/conflict.go

```diff
@@ -166,6 +166,13 @@ func CheckChangeConflictMany(st *state.State, instanceNames []string, ignoreChan
 		if chg == nil || chg.Status().Ready() {
 			continue
 		}
+
+		// task doesn't conflict if it can't run
+		switch task.Status() {
+		case state.HoldStatus, state.ErrorStatus, state.DoneStatus, state.UndoneStatus:
+			continue
```

Comment:
I would move this though just before the affectedSnaps code below

Reply:
I haven't change this code in this branch but I opened another PR for this fix (which incorporates this feedback) with proper tests. Once that merges, I'll just rebase this on it and drop this commit

Reply:
https://github.com/snapcore/snapd/pull/10767